src/libcpu/src/interpreter/interpreter_pairedsingle.cpp

```diff
@@ -559,7 +559,12 @@ ps_res(cpu::Core *state, Instruction instr)
       state->fpr[instr.frD].paired1 = extend_float(d1);
    }
 
+   // ps_res never sets FPSCR[XX], so avoid setting it here.
+   const int xx = state->fpscr.xx;
    updateFPSCR(state, oldFPSCR);
+   if (!xx) {
```

Comment:
According to the documentation, XX is not cleared, it is just untouched.  This should be:
```c++
state->fpsrc.xx = xx;
```